File: fastapi/app/database/models/resonator_master.py
from sqlalchemy import Integer, String
from sqlalchemy.orm import Mapped, mapped_column
from app.database.connection import Base


class ResonatorMaster(Base):
    __tablename__ = "resonator_master"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    name: Mapped[str] = mapped_column(String(50), nullable=False)
    element: Mapped[str] = mapped_column(String(20), nullable=False)
    rarity: Mapped[int] = mapped_column(Integer, nullable=False)
    hp: Mapped[int] = mapped_column(Integer, nullable=False)
    attack: Mapped[int] = mapped_column(Integer, nullable=False)
    defense: Mapped[int] = mapped_column(Integer, nullable=False)
    release_version: Mapped[int] = mapped_column(Integer, nullable=False)
    thumbnail_image: Mapped[str] = mapped_column(String(255), nullable=False)
    standing_image: Mapped[str] = mapped_column(String(255))

File: spring/src/main/java/io/github/wavesync/entity/ResonatorMaster.java
package io.github.wavesync.entity;
import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;
import lombok.*;



@Entity
@Table(name = "resonator_master")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class ResonatorMaster {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 20, unique = true)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Element element;

    @Column(nullable = false)
    private Integer rarity;

    @Column(nullable = false)
    private Integer hp;

    @Column(nullable = false)
    private Integer attack;

    @Column(nullable = false)
    private Integer defense;

    @Column(name = "release_version", nullable = false)
    private Integer releaseVersion;

    @Column(name = "thumbnail_image", nullable = false, length = 255)
    private String thumbnailImage;

    @Column(name = "standing_image", length = 255)
    private String standingImage;

    @OneToOne(mappedBy = "resonatorMaster")
    private ResonanceNodeMaster resonanceNodeMaster;

    @Builder.Default
    @OneToMany(mappedBy = "resonatorMaster")
    private List<UserResonator> userResonators = new ArrayList<>();
}